shared/fs/common/static-breadcrumb.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+import * as React from 'react'
+import {globalStyles, globalColors, globalMargins} from '../../styles'
+import {Box, Icon, Text, type IconType} from '../../common-adapters'
+
+const getIcon = (tlfType: string): IconType => {
+  switch (tlfType) {
+    case 'private':
+      return 'icon-folder-private-16'
+    case 'public':
+      return 'icon-folder-public-16'
+    case 'team':
+      return 'icon-folder-team-16'
+    default:
+      return 'iconfont-question-mark'
+  }
+}
+
+type Props = {
+  pathElements: Array<string>,
+  showTlfTypeIcon?: boolean,
+  includeLast?: boolean,
+}
+
+const StaticBreadcrumb = ({pathElements, showTlfTypeIcon, includeLast}: Props) => (
+  <Box style={stylesBox}>
+    {[
+      showTlfTypeIcon && (
+        <Icon
+          type={getIcon(pathElements[1])}
+          color={globalColors.blue}
+          style={stylesIconFolderType}
+          key="icon"
+        />
+      ),
+      <Text key="text" type="BodySmallSemibold">
+        {pathElements[1]}
+      </Text>,
+      ...pathElements.slice(2, includeLast ? undefined : pathElements.length - 1).map((elem, idx) => [
+        <Icon
+          key={`icon-${idx}`}
+          type="iconfont-arrow-right"
+          style={stylesIconArrow}
+          color={globalColors.black_20}
+          fontSize={12}
+        />,
+
+        // TODO: make this into a component.
+        // We are splitting on ',' here, so it won't work for
+        // long names that don't have comma. If this becomes a
+        // problem, we might have to do smarter splitting that
+        // involve other characters, or just break the long name
+        // apart into 3-character groups.
```

Comment:
This seems to only handle simple comma-separated TLF names. Not conflicting names, and not subdirectories. And it'll split subdirectories on commas. Is this really the desired behavior?

Reply:
As explained in the comment, yes this only deals with comma for now, which is the most common case. We can add other stuff later (or make it fancier) if they become a problem. Do conflicting names not have any space inside? If they have spaces then it's fine. I'm not sure what you mean by subdirectories. It just needs to split long names into chunks so it can wrap properly.

This approach was because we failed to make CSS work for long name wrapping, and Nojima proposed the splitting [here](https://github.com/keybase/client/pull/12314)